ESSArch_Core/test_util.py

```diff
@@ -173,3 +177,155 @@ def test_get_schema_from_file_with_bad_input(self):
+
+    def test_list_files_tarfile_with_default_args_should_return_response(self):
+        file_path = self.create_archive_file('tar')
+        resp = list_files(file_path)
+        self.assertEqual(resp.status_code, 200)
+        file_names = ["./0.txt", "./1.txt", "./2.txt"]  # TODO: bug in shutil for tar is adding an extra './'
+
+        for el in resp.data:
+            data_name = el['name']
+            data_type = el['type']
+            data_size = el['size']
+            data_modified = el['modified']
+
+            self.assertIn(data_name, file_names)
+            file_names.remove(data_name)
+            self.assertEqual(data_type, "file")
+            self.assertEqual(data_size, 1)
+            self.assertEqual(type(data_modified), datetime.datetime)
+
+    def test_list_files_zip_file_with_default_args_should_return_response(self):
+        file_path = self.create_archive_file('zip')
+        resp = list_files(file_path)
+        self.assertEqual(resp.status_code, 200)
+        file_names = ["0.txt", "1.txt", "2.txt"]
+
+        for el in resp.data:
+            data_name = el['name']
+            data_type = el['type']
+            data_size = el['size']
+            data_modified = el['modified']
+
+            self.assertIn(data_name, file_names)
+            file_names.remove(data_name)
+            self.assertEqual(data_type, "file")
+            self.assertEqual(data_size, 1)
+            self.assertEqual(type(data_modified), datetime.datetime)
+
+    def test_list_files_path_to_file_in_tar(self):
+        file_path = self.create_archive_file('tar')
+        new_folder = os.path.join(file_path, "./0.txt")  # TODO: bug in shutil for tar is adding an extra './'
+
+        resp = list_files(new_folder)
+        self.assertEqual(resp.status_code, 200)
+        self.assertEqual(resp['Content-Disposition'], 'inline; filename="0.txt"')
+        self.assertEqual(resp['Content-Type'], 'text/plain; charset=utf-8')
+        self.assertEqual(resp['Cache-Control'], 'no-cache, no-store, must-revalidate')
+        self.assertEqual(resp['Pragma'], 'no-cache')
+        self.assertEqual(resp['Expires'], '0')
+
+    def test_list_files_path_to_non_existing_file_in_tar_should_throw_NotFound(self):
+        file_path = self.create_archive_file('tar')
+        new_folder = os.path.join(file_path, "non_existing_file.txt")
+
+        with self.assertRaises(NotFound):
+            list_files(new_folder)
+
+    def test_list_files_path_to_non_existing_file_in_zip_should_throw_NotFound(self):
+        file_path = self.create_archive_file('zip')
+        new_folder = os.path.join(file_path, "non_existing_file.txt")
+
+        with self.assertRaises(NotFound):
+            list_files(new_folder)
+
+    def test_list_files_path_to_file_in_zip(self):
+        file_path = self.create_archive_file('zip')
+        new_folder = os.path.join(file_path, "0.txt")
+
+        resp = list_files(new_folder)
+
+        self.assertEqual(resp.status_code, 200)
```

Comment:
This is now testing the result of `generate_file_response`. Instead, mock `generate_file_response` and ensure that it is called with the correct arguments. See the following tests for an example: https://github.com/ESSolutions/ESSArch_Core/blob/b01689ebcc2faddbd32664039563f0e77622cb62/ESSArch_Core/WorkflowEngine/tests/test_tasks.py#L37